crypto.go

```diff
@@ -10,29 +10,41 @@ import (
 	"golang.org/x/crypto/nacl/secretbox"
 )
 
-func keyedHash(key, text []byte) []byte {
+func keyedHash(key *[hashLen]byte, text []byte) *[hashLen]byte {
 	h := keyedHasher(key)
 	h.Write([]byte(text))
-	return h.Sum(nil)
+	var sum [hashLen]byte
+	r := h.Sum(sum[:0])
+	if &r[0] != &sum[0] {
```

Comment:
I had a sense this was what was going on, just wanted to be sure. Thanks for the explanation!